web/src/rendered_markdown.js

```diff
@@ -139,11 +139,22 @@ export const update_elements = ($content) => {
             // assuming that you're not searching for text
             // inside the highlight.
             const person = people.maybe_get_user_by_id(user_id, true);
-            if (person !== undefined) {
+            if (person === undefined || person.is_inaccessible_user) {
                 // Note that person might be undefined in some
-                // unpleasant corner cases involving data import.
-                set_name_in_mention_element(this, person.full_name, user_id);
+                // unpleasant corner cases involving data import
+                // or when guest users cannot access all users in
+                // the organization.
+                // For such cases we set the name to be the same as
+                // in the message content received from server and
+                // the "Unknown user" popover is opened on clicking
+                // the mention pill.
+                if (person === undefined) {
+                    people.add_inaccessible_user(user_id);
```

Comment:
I am not sure how to handle the case when the markdown element only has `data-user-email` and not `data-user-id`. I don't think we would want to create fake user objects with no or some random ID. Or do we just add separate code to show the "Unknown user" in the click handler, without creating any user object.

Reply:
Such cases are super rare and very old (1137ef4e40258280e202325c9295579c93fb948b). All we need is for the client to not crash on displaying them -- It's probably fine if clicking on such an ancient mention doesn't open the popover, for example -- that will realistically probably never happen again for those 6+ year old mentions.